keras/preprocessing/image.py

```diff
@@ -437,6 +437,50 @@ def fit(self, X,
             U, S, V = linalg.svd(sigma)
             self.principal_components = np.dot(np.dot(U, np.diag(1. / np.sqrt(S + 10e-7))), U.T)
 
+    def transform(self,x, train_mode):
+        '''
+        Transforms single image.
+        Required for featurewise_center, featurewise_std_normalization
+        and zca_whitening.
+
+        # Arguments
+            X: Numpy array, the data to fit on.
+            augment: whether to fit on randomly augmented samples
+            rounds: if `augment`,
+                how many augmentation passes to do over the data
+            seed: random seed.
+
+        Arguments:
+            x: image(Numpy array) to transform
+            train_mode: True if random augmentation should be applied, False - steps needed to be applied on tst and train data(aka normalisation)
+
+        Returns: transformed image
+        '''
+        if train_mode:
+            x = self.random_transform(x)
+        return self.standardize(x)
```

Comment:
what if I change train_mode to random_transform? Or you think there should not be any case where you need only one of them. I think single mode, batch mode and generator mode still can  be useful with/without train_mode(or other name) param.